colossalai/zero/gemini/gemini_ddp.py

```diff
@@ -327,7 +335,11 @@ def grad_handle(self, p, grad):
                 )
             grad_chunk = chunk
             if not self.reuse_fp16_chunk:
-                grad_chunk = self.chunk_manager.init_grad_chunk(chunk)
+                if not self.accumulating_grads:
+                    grad_chunk = self.chunk_manager.init_grad_chunk(chunk)
+                else:
+                    self.chunk_manager.access_chunk(chunk.grad_chunk)
```

Comment:
This leads to unnecessary all-gather. Can we avoid this?

Reply:
modified